Fix configure script to work with targets like 'sh4-unknown-linux-gnu'

--- configure.ac
+++ configure.ac
@@ -91,14 +91,14 @@
 	arch=hppa
 	AC_DEFINE([HPPA], 1, [Define for the HPPA architecture.])
 	;;
-sh)
-	arch=sh
-	AC_DEFINE([SH], 1, [Define for the SH architecture.])
-	;;
-sh64)
+sh64*)
 	arch=sh64
 	AC_DEFINE([SH64], 1, [Define for the SH64 architecture.])
 	;;
+sh*)
+	arch=sh
+	AC_DEFINE([SH], 1, [Define for the SH architecture.])
+	;;
 x86?64*)
 	arch=x86_64
 	AC_DEFINE([X86_64], 1, [Define for the AMD x86-64 architecture.])
